scipy/spatial/transform/_rotation.pyx

```diff
@@ -1187,10 +1218,27 @@ cdef class Rotation:
         (2, 4)
 
         """
+        quat = self._quat
+        cdef double x, y, z, w
+        if order == 'wxyz':
+            quat = self._quat.copy()
+            # Convert from internal 'xyzw' format
+            for ind in range(len(quat)):
+                x = quat[ind, 0]
+                y = quat[ind, 1]
+                z = quat[ind, 2]
+                w = quat[ind, 3]
+                quat[ind, 0] = w
+                quat[ind, 1] = x
+                quat[ind, 2] = y
+                quat[ind, 3] = z
```

Comment:
is there any value in noticing the equivalence to `np.roll(quat)`? I was just thinking it might be more concise and there is already a `copy` happening in the `return` statements anyway.

Something like `return np.array(np.roll(quat), copy=True)`  ?

I suppose for the opposite conversion you can use the negative `shift` for roll as well. Just a thought, maybe there's a reason you didn't do that.

Reply:
Great suggestion.
I was trying to do slices, which doesn't work in Cython. I didn't think to use `np.roll`.